### PR TITLE
[Feature] DETERMINISTIC interaction mode

### PR DESCRIPTION
**Pull request by vmoens**
cc @matteobettini

### CODE REVIEW

File: tensordict/nn/probabilistic.py [hunk 1]
```diff
@@ -453,8 +454,26 @@ def _dist_sample(
             interaction_type = self.default_interaction_type
 
         if interaction_type is InteractionType.MODE:
```
Comment (matteobettini) [inline tensordict/nn/probabilistic.py:456]:
```suggestion
        if interaction_type is InteractionType.DETERMINISTIC:
```

File: tensordict/nn/probabilistic.py [hunk 2]
```diff
@@ -453,8 +454,26 @@ def _dist_sample(
...
+                    warnings.warn(
+                        "deterministic_sample wasn't found when queried. "
+                        f"{type(self).__name__} is falling back on mean instead. "
+                        f"For better code quality and efficiency, make sure to either "
+                        f"provide a distribution with a deterministic_sample attribute or "
+                        f"to change the InteractionMode to the desired value.",
+                        category=UserWarning,
+                    )
+
+        if interaction_type is InteractionType.MODE:
```
Comment (matteobettini) [inline tensordict/nn/probabilistic.py:476]:
```suggestion
        elif interaction_type is InteractionType.MODE:
```

File: tensordict/nn/probabilistic.py [hunk 3]
```diff
@@ -453,8 +454,26 @@ def _dist_sample(
...
-            if hasattr(dist, "get_mode"):
-                return dist.get_mode()
+            try:
+                return dist.deterministic_sample
+            except AttributeError:
+                try:
+                    return dist.mean
+                except AttributeError:
+                    raise NotImplementedError(
+                        f"method {type(dist)}.mode is not implemented."
```
Comment (matteobettini) [inline tensordict/nn/probabilistic.py:464]: maybe say you asked deterministic and we fall back to mean and did not find it

Comment (vmoens) [reply tensordict/nn/probabilistic.py:464]:
```suggestion
                        f"method {type(dist)}.deterministic_sample is not implemented."
```

Comment (vmoens) [reply tensordict/nn/probabilistic.py:464]: @matteobettini it's in the finally below

File: tensordict/nn/probabilistic.py (hunk 2 above)
Comment (vmoens) [reply tensordict/nn/probabilistic.py:476]: not necessary, the function either returns or raises just before

**Pull request merged**